proposals/ImplicitCopyWarnings/LinearRegression-warnings.txt

```diff
@@ -0,0 +1,84 @@
+reduced.swift:14:23: warning: 'inputs' implicitly copied to the accelerator, use .toAccelerator() to make transfer explicit
+  public func predict(inputs: Tensor<Float>) -> Tensor<Float> {
```

Comment:
hmm i don't understand why such warnings appear in the first place -- there should be no sends/recvs here.
maybe this is due to the use of `mutating`?

Reply:
This is because `predict(inputs:)` is being partitioned as a deabstraction scope. It's a public function.

Reply:
Before we have a better inlining model, should we recommend that all public APIs be marked as "inline always"? Otherwise if user code calls a sequence of public APIs, that could generate a lot of sends/recvs.

Reply:
I think this discussion is mostly orthogonal to the warnings proposal. Regardless of what causes such a function to get partitioned on its own, we need to make the warnings good when it does get partitioned.

Reply:
Yep, it is orthogonal! But at the same time I hope this is providing some context on how argument send/receive warnings are emitted in the first place.

One middle ground I can think of is to have the compiler still partition inlinable functions outside the TF module. This would resolve both Mingsheng’s and mine concerns.

Reply:
Richard's proposal SGTM. 

I agree with Mark that what's partitionable is itself orthogonal to the warnings proposal. One related aspect is: when we are partitioning a public API, but if it does not get called in user code, it'd be nice not to generate sends/recvs warnings for it?